pkg/kubelet/nodelease/controller_test.go

```diff
@@ -282,3 +282,111 @@ func TestRetryUpdateLease(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateUsingLatestLease(t *testing.T) {
+	node := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+			UID:  types.UID("foo-uid"),
+		},
+	}
+
+	gr := schema.GroupResource{Group: "v1", Resource: "lease"}
+	notFoundErr := apierrors.NewNotFound(gr, "foo")
+	cases := []struct {
+		desc                       string
+		latestLease                *coordinationv1.Lease
+		updateReactor              func(action clienttesting.Action) (bool, runtime.Object, error)
+		getReactor                 func(action clienttesting.Action) (bool, runtime.Object, error)
+		createReactor              func(action clienttesting.Action) (bool, runtime.Object, error)
+		onRepeatedHeartbeatFailure func()
+		expectLeaseResourceVersion string
+	}{
+		{
+			desc:          "latestLease is nil and need to create",
+			latestLease:   nil,
+			updateReactor: nil,
+			getReactor: func(action clienttesting.Action) (bool, runtime.Object, error) {
+				return true, nil, notFoundErr
+			},
+			createReactor: func(action clienttesting.Action) (b bool, object runtime.Object, e error) {
+				return true, &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "1"}}, nil
```

Comment:
Currently it doesn't change much, but I think we should set at least name and namespace.

Sp let's introduce a simple helper function
`makeLease(name, resourceVersion) *coordinationv1.Lease`
and use it here and below

Reply:
done